### PR TITLE
Picker network updates

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**
In the [new designs for Amon Hen](https://github.com/MetaMask/metamask-extension/pull/21220/files#diff-afc7617dc1347d665d4f7a058e9ea8ac686f5ad520874a293f6e27e7c50f9568) the network picker(`PickerNetwork`) has been moved to inside of the tabs and needs to be full width. The current `PickerNetwork` requires some style updates to allow for this change

This PR updates the `PickerNetwork` component styles and props to allow for this updated design. It uses the pattern of inversion of control to allow for the label wrapper component to be accessed through a prop called `labelProps`. It also updates the unit tests to include coverage for this change as well as updates the documentation and stories to provide examples.

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Search `PickerNetwork` 
3. See that there are no visual regressions
4. Check the updated documentation and Width story

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/7ca4f2ea-63fb-4e12-a3ab-34e8890bbbfc

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/22036bca-7ebd-418d-8fae-c15f055bab0c

Other instances of `PickerNetwork` without visual regressions

Screenshot 2023-10-10 at 10 29 55 PMScreenshot 2023-10-10 at 7 54 53 PM

Showing new-network-info modal with some deprecated components replaced and flex box added

Screenshot 2023-10-11 at 3 12 43 PM

Screenshot 2023-10-10 at 9 53 33 PM
Screenshot 2023-10-10 at 10 29 24 PM

Test coverage 100%
Screenshot 2023-10-10 at 6 41 48 PM

## **Related issues**
https://github.com/MetaMask/metamask-extension/pull/21220/files#diff-afc7617dc1347d665d4f7a058e9ea8ac686f5ad520874a293f6e27e7c50f9568

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by MaxwellDG** [comment]
@georgewrmarshall I take this as I should close #20825 #21051 and #21083. When this PR is completed I'll redo them to match the update

File: test/e2e/tests/add-custom-network.spec.js [hunk 1]
```diff
@@ -508,7 +508,7 @@ describe('Custom network', function () {
 
           const arbitrumNetwork = await driver.findElements({
             text: 'Arbitrum One',
-            tag: 'button',
+            tag: 'span',
```
Comment (georgewrmarshall) [inline test/e2e/tests/add-custom-network.spec.js:511]: This might not be correct

File: ui/components/ui/new-network-info/index.scss [hunk 2]
```diff
@@ -1,47 +1,5 @@
...
-  &__manually-add-tokens {
-    display: inline;
-  }
-}
-
-.chip--with-left-icon {
-  padding-left: 8px;
-  padding-top: 8px;
-  padding-bottom: 8px;
 }
```
Comment (georgewrmarshall) [inline ui/components/ui/new-network-info/index.scss:5]: Removing unused styles

File: ui/components/ui/new-network-info/new-network-info.js [hunk 3]
```diff
@@ -69,155 +76,74 @@ export default function NewNetworkInfo() {
...
-        </Box>
-      </Popover>
+          <Button onClick={onCloseClick} size={ButtonSize.Lg} block>
+            {t('recoveryPhraseReminderConfirm')}
+          </Button>
+        </ModalContent>
+      </Modal>
     )
   );
 }
```
Comment (georgewrmarshall) [inline ui/components/ui/new-network-info/new-network-info.js:149]: ## Before

Screenshot 2023-10-10 at 7 55 37 PM


### After

Screenshot 2023-10-10 at 6 42 23 PM


https://github.com/MetaMask/metamask-extension/assets/8112138/6962ef26-ec24-4be9-8692-82e3b8d5bf7e

File: ui/components/ui/new-network-info/new-network-info.stories.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import NewNetworkInfo from '.';
+
+export default {
+  title: 'Components/UI/NewNetworkInfo',
+};
+
+export const DefaultStory = () => <NewNetworkInfo />;
+
+DefaultStory.storyName = 'Default';
```
Comment (georgewrmarshall) [inline ui/components/ui/new-network-info/new-network-info.stories.js:10]: Adding storybook file for new-network-info component for easy review

File: ui/components/ui/new-network-info/new-network-info.js [hunk 5]
```diff
@@ -9,15 +9,15 @@ import {
...
   TextVariant,
 } from '../../../helpers/constants/design-system';
 import { IMPORT_TOKEN_ROUTE } from '../../../helpers/constants/routes';
 import { getCurrentNetwork, getUseTokenDetection } from '../../../selectors';
 import { setFirstTimeUsedNetwork } from '../../../store/actions';
-import { PickerNetwork, Text } from '../../component-library';
-import Box from '../box';
+import { PickerNetwork, Text, Box } from '../../component-library';
 import Button from '../button';
 import Popover from '../popover';
```
Comment (georgewrmarshall) [inline ui/components/ui/new-network-info/new-network-info.js:22]: Will replace these in another PR

File: test/e2e/tests/add-custom-network.spec.js [hunk 6]
```diff
@@ -458,7 +458,7 @@ describe('Custom network', function () {
           });
           // verify network switched
           const networkDisplayed = await driver.findElement({
-            tag: 'p',
+            tag: 'span',
```
Comment (georgewrmarshall) [inline test/e2e/tests/add-custom-network.spec.js:461]: Changed the wrapping html element of the text from a `p` to `span` tag in the `PickerNetwork`

File: test/e2e/tests/custom-rpc-history.spec.js [hunk 7]
```diff
@@ -70,7 +70,7 @@ describe('Stores custom RPC history', function () {
           '.networks-tab__add-network-form-footer .btn-primary',
         );
 
-        await driver.findElement({ text: networkName, tag: 'p' });
+        await driver.findElement({ text: networkName, tag: 'span' });
```
Comment (georgewrmarshall) [inline test/e2e/tests/custom-rpc-history.spec.js:73]: Changed the wrapping html element of the text from a `p` to `span` tag in the `PickerNetwork`

**Review by darkwing — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

File: test/e2e/tests/switch-custom-network.spec.js [hunk 8]
```diff
@@ -88,7 +88,7 @@ describe('Switch ethereum chain', function () {
         await driver.switchToWindow(extension);
 
         const currentNetworkName = await driver.findElement({
-          tag: 'p',
+          tag: 'span',
```
Comment (georgewrmarshall) [inline test/e2e/tests/switch-custom-network.spec.js:91]: Changed the wrapping html element of the text from a `p` to `span` tag in the `PickerNetwork`

**Comment by MaxwellDG** [comment]
> @georgewrmarshall I take this as I should close #20825 #21051 and #21083. When this PR is completed I'll redo them to match the update

@georgewrmarshall

**Review by darkwing — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**